pkg/oci/utils.go

```diff
@@ -161,57 +196,115 @@ func networkConfig(ocispec CompatOCISpec) (vc.NetworkConfig, error) {
-		Interactive:  ocispec.Process.Terminal,
-		Console:      console,
+	for _, podStatus := range podStatusList {
+		for _, containerStatus := range podStatus.ContainersStatus {
+			if containerStatus.ID == containerID {
+				return containerStatus, podStatus.ID, nil
+			}
+		}
 	}
 
-	cmd.SupplementaryGroups = []string{}
-	for _, gid := range ocispec.Process.User.AdditionalGids {
-		cmd.SupplementaryGroups = append(cmd.SupplementaryGroups, strconv.FormatUint(uint64(gid), 10))
+	return vc.ContainerStatus{}, "", fmt.Errorf("Could not find any pod/container ID %s", containerID)
+}
+
+// GetContainerType determines which type of container matches the annotations
+// table provided.
+func GetContainerType(annotations map[string]string) (ContainerType, error) {
+	if containerType, ok := annotations[ContainerTypeKey]; ok {
+		return ContainerType(containerType), nil
 	}
 
-	containerConfig := vc.ContainerConfig{
-		ID:             cid,
-		RootFs:         rootfs,
-		ReadonlyRootfs: ocispec.Spec.Root.Readonly,
-		Cmd:            cmd,
-		Annotations: map[string]string{
-			ConfigPathKey: configPath,
-			BundlePathKey: bundlePath,
-		},
+	ociLog.Errorf("Annotations[%s] not found, cannot determine the container type",
+		ContainerTypeKey)
+	return UnknownContainerType, fmt.Errorf("Could not find container type")
+}
+
+// GetContainerTypeFromConfig returns the type of container based on CRI-O
+// annotations.
+func GetContainerTypeFromConfig(ociSpec CompatOCISpec) (ContainerType, error) {
+	containerType, ok := ociSpec.Annotations[CRIOContainerTypeKey]
+	if !ok {
+		ociLog.Infof("Annotations[%s] not found, this is a regular container",
+			CRIOContainerTypeKey)
+		return RegularContainer, nil
+	}
+
+	switch containerType {
+	case ContainerTypePod:
+		return PodSandbox, nil
+	case ContainerTypeContainer:
+		return PodContainer, nil
+	}
+
+	return UnknownContainerType, fmt.Errorf("Unknown container type %s", containerType)
+}
+
+// IsPod determines if the container type can be considered as a pod.
+// We can consider a pod in case we have a PodSandbox or a RegularContainer.
+func IsPod(cType ContainerType) bool {
+	if cType == PodSandbox || cType == RegularContainer {
+		return true
+	}
+
+	return false
+}
+
+// PodContainerPodID determines the pod ID related to an OCI configuration from a specific
+// annotation. This function is expected to be called only when the container
+// type is "PodContainer".
+func PodContainerPodID(annotations map[string]string) (string, error) {
+	podID, ok := annotations[CRIOSandboxNameKey]
+	if !ok {
+		return "", fmt.Errorf("Could not find pod ID: Unknown Annotations[%s]", CRIOSandboxNameKey)
 	}
 
+	return podID, nil
+}
+
+// PodConfig converts an OCI compatible runtime configuration file
+// to a virtcontainers pod configuration structure.
+func PodConfig(ocispec CompatOCISpec, runtime RuntimeConfig, bundlePath, cid, console string) (vc.PodConfig, error) {
```

Comment:
Fair enough. I feel the API is a bit redundant now, but it's for the good cause.